src/mono/mono/component/debugger-agent.c

```diff
@@ -8846,7 +8846,10 @@ type_commands_internal (int command, MonoClass *klass, MonoDomain *domain, guint
 	}
 	case MDBGPROT_CMD_TYPE_ELEMENT_TYPE: 
 	{
-		buffer_add_int (buf, m_class_get_byval_arg (klass)->type);
+		if (m_class_get_byval_arg (klass)->type == MONO_TYPE_GENERICINST && MONO_TYPE_ISSTRUCT (m_class_get_byval_arg (klass)))
```

Comment:
Please add a comment that this is intentional behavior for compat with coreclr mscordbi.  `MDBGPROT_CMD_TYPE_ELEMENT_TYPE` sounds like a general-purpose command, but it's really kind of weird.